### PR TITLE
Add option to control rechunking in reshape

### PR DESCRIPTION
**Pull request by TomAugspurger**
Builds on top of #6748, so a WIP for now. Just pushing this to show the docs to aid in review over there.

In `da.reshape`, there's a tradeoff between

1. The chunking of "slow-moving" axes (specifically whether they all have chunksize=1)
2. The amount of communication / merging done in the reshape

This adds a keyword to control that tradeoff. The default `merge_chunks=True` is the current behavior and is backwards compatible.

Here's a render of the new docs

array-reshaping

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Here's the gist / nbviewer used to make the plots: https://gist.github.com/36422ec34e1c07c1c98a3a5aed3ef860 / https://nbviewer.jupyter.org/gist/TomAugspurger/36422ec34e1c07c1c98a3a5aed3ef860.

File: dask/array/reshape.py [hunk 1]
```diff
@@ -143,9 +143,24 @@ def contract_tuple(chunks, factor):
...
     """Reshape array to new shape
 
+    Parameters
+    ----------
+    shape : int or tuple of ints
+        The new shape should be compatible with the original shape. If
+        an integer, then the result will be a 1-D array of that length.
+        One shape dimension can be -1. In this case, the value is
+        inferred from the length of the array and remaining dimensions.
+    merge_chunks : bool, default True
```
[the remark this answers is not in the document]
Comment (jsignell) [reply dask/array/reshape.py:156]: I like it flipped, but `rechunk_leading_axes` feels long. This feels kind of like `split_every=1` or something, but maybe reusing that kwarg is too confusing. Maybe `split_chunks` would work?

**Comment by jsignell** [comment]
I am fine with merging this as-is and seeing how it goes. Does that work for you @TomAugspurger?

**Comment by TomAugspurger** [comment]
Works for me :)

**Pull request merged**